### DISCUSSION TITLE
Zabbix / Joomla Monitoring

### DISCUSSION THREAD

**Discussion by bitts** [joomla/joomla-cms#42202] [2. Cold Features]
As I manage hosting sites that use Joomla and need to constantly monitor several of these platforms, I created a template using only PHP (https://github.com/bitts/zabbix_joomla) for Zabbix (https://github.com/zabbix/zabbix) which is a tool for monitoring networks, servers and services and collects a lot of data from this platform, making work easier, creating alerts or trigger actions according to the data collected. I currently check 16 items, including: number of users and their access levels, number of backup files, system version, folder size, among other items.
This monitor is compatible with versions up to 3.x and I have not carried out tests with versions above that and adjustments may be necessary for perfect operation in versions above 4.x.
In addition to sharing this monitoring, which helps me a lot, I would like opinions on the subject and tips on other items that can and should be monitored in Joomla. If you are interested in helping me improve or correct problems with this script, I would be very grateful and I believe it has a lot to help the community as a whole!

**Comment by Akshat111111** [comment]
It's fantastic that you've created a Joomla monitoring template utilizing PHP and Zabbix! Monitoring and alerting are critical for keeping web platforms healthy and secure. Your contribution to the community is significant.

You've covered several crucial features of extra Joomla things to monitor, such as the amount of users, access permissions, backups, and version.
But It will be good and helpful if you-
1)Think about connecting Joomla with technologies such as Google Analytics to track website traffic, page views, and user interactions.
2)Watch for changes in user roles and permissions, as well as changes in user login and logout activities. Unusual user behavior may pose a security risk.
3)Check the error logs of Joomla for any PHP problems, warnings, or significant issues. This can assist in identifying and addressing issues before they have an impact on the site's functioning.

**Comment by bitts** [comment]
I stuck to PHP precisely because it is Joomla's native language, which I believe would make compatibility, maintenance and monitoring much easier for me, the community or those who master this incredible tool!  Thank you very much for your feedback... I'm going to start studying Google Analytics to find out how to integrate it into monitoring in a simple and practical way and I feel silly for not thinking about keeping an eye on the Joomla error logs for any PHP problems (really something essential) and which is not complex to do, which I intend to resolve as soon as possible!
